### ISSUE TITLE
Assertion pindexPrev && pindexPrev == chainstate.m_chain.Tip() when running regtest

### ISSUE THREAD

**Issue by dkatzan** [bitcoin/bitcoin#31562]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

I'm running a regtest chain in my CI\CD for some tests, occasionally, I will see my bitcoind fail with an assertion 
`Assertion failed: pindexPrev && pindexPrev == chainstate.m_chain.Tip() (validation.cpp: TestBlockValidity: 4338)`

### Expected behaviour

The node to be stable and not crash

### Steps to reproduce

I don't have steps to reproduce, this happens very rearily as part of our CI\CD that runs a simple regtest chains, mines a few blocks, makes a few transactions and managing a few wallets

### How did you obtain Bitcoin Core

Package manager

### What version of Bitcoin Core are you using?

27.0-r1

### Operating system and version

alpine:3.18

### Machine specifications

I'm building an alpine image version 3.18
and installing bitcoin in my docker file as such
```
RUN apk add bitcoin=25.1-r1 --repository=https://dl-cdn.alpinelinux.org/alpine/v3.19/community
RUN apk add bitcoin-cli
```
since 25 is now a bit old, I tried bumping the version, so I'm now running 
```
RUN apk add bitcoin=25.1-r1 --repository=https://dl-cdn.alpinelinux.org/alpine/v3.19/community
RUN apk add bitcoin-cli
```
but the issue persisted

**Comment by dkatzan** [comment]
Yep, I'm calling `generateblock` multiple times
my tests are also running concurrently, so it might be called concurrently

here are the latest logs I have from before the crash
[extract-2024-12-24T14_43_27.491Z.csv](https://github.com/user-attachments/files/18240488/extract-2024-12-24T14_43_27.491Z.csv)

**Comment by dkatzan** [comment]
Thx alot!
so is this expected to reach the next release cycle? which version will it be included in?

**Comment by ismaelsadeeq** [comment]
>  so is this expected to reach the next release cycle? which version will it be included in?

After it is reviewed and merged yes it will be available in the next release. 
You can check #31029 for the plans regarding the next release.  
You can also volunteer to test and ensure that #31563 addresses your issue.

**Comment by mzumsande** [comment]
> Thx alot! so is this expected to reach the next release cycle? which version will it be included in?

Looks like it will already be fixed in 28.1 (backport PR #31594)  which should be released soon.
